azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -1255,4 +1267,179 @@ public void accept(MessageSender a) {
+				@Override
+				public CompletableFuture<EventHubPartitionRuntimeInformation> apply(Map<String, Object> rawdata) {
+					CompletableFuture<EventHubPartitionRuntimeInformation> future2 = new CompletableFuture<EventHubPartitionRuntimeInformation>();
+					future2.complete(new EventHubPartitionRuntimeInformation(
+							(String)rawdata.get(ClientConstants.MANAGEMENT_ENTITY_NAME_KEY),
+							(String)rawdata.get(ClientConstants.MANAGEMENT_PARTITION_NAME_KEY),
+							(long)rawdata.get(ClientConstants.MANAGEMENT_RESULT_BEGIN_SEQUENCE_NUMBER),
+							(long)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_SEQUENCE_NUMBER),
+							(String)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_OFFSET),
+							((Date)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_TIME_UTC)).toInstant()));
+					return future2;
+				}
+	        });
+        }
+        
+        return future1;
+    }
+    
+    private <T> CompletableFuture<T> addManagementToken(Map<String, String> request)
+    {
+    	CompletableFuture<T> retval = null;
+        try {
+        	String audience = String.format("amqp://%s/%s", this.underlyingFactory.getHostName(), this.eventHubName);
+        	String token = this.underlyingFactory.getTokenProvider().getToken(audience, ClientConstants.TOKEN_REFRESH_INTERVAL); 
+			request.put(ClientConstants.MANAGEMENT_SECURITY_TOKEN_KEY, token);
+		} 
+        catch (InvalidKeyException | NoSuchAlgorithmException | IOException e) {
+        	retval = new CompletableFuture<T>();
+        	retval.completeExceptionally(e);
+		}
+    	return retval;
+    }
+    
+    private CompletableFuture<Map<String, Object>> managementWithRetry(Map<String, String> request) {
+        Instant endTime = Instant.now().plus(this.underlyingFactory.getOperationTimeout());
+        CompletableFuture<Map<String, Object>> rawdataFuture = new CompletableFuture<Map<String, Object>>();
+        
+        ManagementRetry retrier = new ManagementRetry(rawdataFuture, endTime, this.underlyingFactory, request, 0, null);
+        Timer.schedule(retrier, Duration.ZERO, TimerType.OneTimeRun);
+        
+        return rawdataFuture;
+    }
+    
+    private class ManagementRetry implements Runnable {
+    	private final CompletableFuture<Map<String, Object>> finalFuture;
+    	private final Instant endTime;
+    	private final MessagingFactory mf;
+    	private final Map<String, String> request;
+    	private final int retryCount;
+    	private final String clientId;
+    	
+    	public ManagementRetry(CompletableFuture<Map<String, Object>> future, Instant endTime, MessagingFactory mf,
+    			Map<String, String> request, int retryCount, String clientId) {
+    		this.finalFuture = future;
+    		this.endTime = endTime;
+    		this.mf = mf;
+    		this.request = request;
+    		this.retryCount = retryCount;
+    		if (clientId != null) {
+    			this.clientId = clientId;
+    		}
+    		else {
+    			this.clientId = UUID.randomUUID().toString();
```

Comment:
>clientId [](start = 12, length = 8)

lets use this.mf.getMgmtChannel().getClientId() - per our disc. we will anyway have 1 mgmtChannel per EHClient
- this way - retrypolicy on all Mgmt operations would be same